File: src/main/java/uk/co/lukestevens/geoguessr/models/Player.java
package uk.co.lukestevens.geoguessr.models;

import javax.persistence.*;
import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "players", schema = "geoguessr")
public class Player {

    @Id
    @GeneratedValue(strategy=GenerationType.IDENTITY)
    @Column(name = "player_id")
    private Long id;

    @Column(name = "geoguessr_id")
    private String geoguessrId;

    @Column(name = "display_name")
    private String displayName;

    @OneToMany(mappedBy = "player", cascade = CascadeType.ALL, fetch = FetchType.EAGER)
    private List<PlayerScore> playerScores = new ArrayList<>();

    public Player(String geoguessrId, String displayName) {
        this.geoguessrId = geoguessrId;
        this.displayName = displayName;
    }

    public Player() { }

    public Long getId() {
        return id;
    }

    public String getGeoguessrId() {
        return geoguessrId;
    }

    public String getDisplayName() {
        return displayName;
    }

    public List<PlayerScore> getPlayerScores() {
        return playerScores;
    }
}

File: src/main/kotlin/uk/co/lukestevens/geoguessr/models/Player.kt
package uk.co.lukestevens.geoguessr.models

import javax.persistence.CascadeType
import javax.persistence.Column
import javax.persistence.Entity
import javax.persistence.FetchType
import javax.persistence.GeneratedValue
import javax.persistence.GenerationType
import javax.persistence.Id
import javax.persistence.OneToMany
import javax.persistence.Table

@Entity
@Table(name = "players", schema = "geoguessr")
class Player {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "player_id")
    val id: Long? = null

    @Column(name = "geoguessr_id")
    var geoguessrId: String? = null

    @Column(name = "display_name")
    var displayName: String? = null

    @OneToMany(mappedBy = "player", cascade = [CascadeType.ALL], fetch = FetchType.EAGER)
    val playerScores: MutableList<PlayerScore> = ArrayList()

    constructor(geoguessrId: String?, displayName: String?) {
        this.geoguessrId = geoguessrId
        this.displayName = displayName
    }

    constructor()
}